[SHIRO-261] - Improvements to how-to-contribute guide

pitjazz opened the pull request:
Simplified and modified contribute page so that more people would read it and volunteer.

bmarwell commented on how-to-contribute.md:
```diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
-*   Prepend your email subject line with a marker when that is appropriate, e.g. [Proposal], [RT] (Random Thought which quickly blossom into research topics <i class="fa fa-smile-o" aria-hidden="true"></i>, [STATUS] (development status of a certain facility).
-*   Remember that most people are participating in development on a volunteer basis and in their "spare time". These enthusiasts will attempt to respond to issues. It may take a little while to get your answers.
-*   Research your topic thoroughly before beginning to discuss a new development issue. Search and browse through the email archives - your issue may have been discussed before. Do not just perceive a problem and then rush out with a question - instead, delve.
+###Links
```
Space after `#`

bmarwell commented on how-to-contribute.md:
```diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
+*	There are no dumb questions. But browse, search and learn from mailing list archives.
+*	Research your topic thoroughly before beginning to discuss a new development issue. 
+
+###Emails
```
space after `#`

bmarwell commented on how-to-contribute.md:
```diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
-*   When sending a patch, you usually do not need to worry about which Git branch it should be applied to. The maintainers of the repository will decide.
-*   Keep all project-related discussion on the mailing list. It is much better to utilise the wider audience, rather than to break off into private discussion groups. You never know who else will have the answer to your issues, and anyway other people are interested in the outcome.
-*   Become familiar with the mailing lists. As you browse and search, you will see the way other people do things. Follow the leading examples.
+*   Prepend your email subject line with a marker when that is appropriate, e.g.:
+**	[Proposal]
+**	[RT] (Random Thought which quickly blossom into research topics)
+**	[STATUS] (development status of a certain facility)
+
+*   Reduce clutter, don't send an e-mail which simply says "thanks".
+*   Keep all project-related discussion on the mailing list. It is better to utilise the wider audience, rather than to break off into private discussions. You never know who else will have the answer to your issues, and anyway other people are interested in the outcome.
```
while at it, please only one space after `*` or `**`

bmarwell commented on how-to-contribute.md:
```diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
 
+<a name="HowtoContribute-git"></a>
+<a name="HowtoContribute-GitUsage"></a>
+##Git Usage
```
space after  `#`

bmarwell commented on how-to-contribute.md:
```diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
+An overview of how to use Git to participate in Shiro development. Beginners, do't be afraid - you cannot accidentally destroy the actual code repository, because you are working with a local copy as an anonymous user. Therefore, you do not have the system permissions to change anything. You can only update your local repository and compare your revisions with the real repository. The [Download S ...
+
 <a name="HowtoContribute-gettingStartedGit"></a>
-###Getting Started
+###Git, Getting Started
```
space after `#`, maybe "getting started with git".
If this is "getting started with git", where is "getting started with code/development/etc"?

bmarwell commented on how-to-contribute.md:
```diff
@@ -50,51 +68,28 @@ The Roadmap is the best way to get an overview. The Unscheduled list also needs
 
 <a name="HowtoContribute-submitThroughJIRA"></a>
-###Submitting a patch through JIRA
+###Submitting a Patch Through JIRA
```
why the capitalization?

bmarwell reviewed (changes requested):
Thanks for your contribution!
Could you please squash your commits and take a look at some nits I found?

Thanks!

bdemers commented on how-to-contribute.md:
```diff
@@ -30,17 +44,21 @@ Document writers are usually the most wanted people so if you like to help but y
 <a name="HowtoContribute-procedure"></a>
 <a name="HowtoContribute-Procedureforreportingbugsandissuesandenhancementsuggestions"></a>
-##Procedure for reporting bugs and issues and enhancement suggestions
+##Procedure for Reporting Bugs and Issues and Enhancement Suggestions
```
It's a common pattern (and assumes that header sections are treated as titles), InfoQ actually has a cool tool to help with this 😄 
http://www.conissaunce.com/pages/infoq-title-case.html

bmarwell replied:
Oh I didn’t know headline case was a thing! Good thing to have a native speaker in the team! :) 
I only knew sth similar from music brainz. ;-)

pitjazz commented on how-to-contribute.md:
````diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
 $ git config --global user.name "Your Name"
 $ git config --global user.email you@domain.com
 ```
 
-You'll grab the Shiro source with git:
+Grab the Shiro source:
 
 ``` bash
 $ git clone https://gitbox.apache.org/repos/asf/shiro.git
 ```
 
-If you already have the source, make sure you're working with the most recent version. Do a `git pull` if you cloned the source more than a few hours ago. (Apache Shiro development can move pretty fast!)
+Make sure you're working with the most recent version. Do a `git pull` if you cloned the source more than a few hours ago.
 
 ``` bash
 $ git checkout -b mybranch
 ```
````
Squashed the reverting commit.

Added instruction to fork in "getting started" section.

pitjazz commented on how-to-contribute.md:
````diff
@@ -104,74 +99,101 @@ If you want to add a feature, you should bring it up for discussion on the dev@s
 ```
 
-Make whatever changes you're going to make, be sure to use git add to stage the changes, and then you're going to commit the changes to your working branch:
+Use `git add` to stage the changes. Commit the changes to your working branch:
````
Do you mean the capitalization this phrase: "Use `git add` to stage the changes. Commit the changes to your working branch:"? I think it should be correct as it is...

pitjazz commented:
Please take a look to pull request again. Modifications made based on comments.

pitjazz commented:
@bdemers, please take a look at this review?

bdemers commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
+* general hardware/money [donations](http://www.apache.org/foundation/thanks.html) via the [Apache Software Foundation](http://www.apache.org)
 
-You can get your local working copy of the [latest and greatest code](download.html "Download") by following the directions in our [Download](download.html "Download") page. Review the To Do list in the [issue tracker](https://issues.apache.org/jira/browse/SHIRO) and then choose a task that interests you. Perhaps you have noticed something that needs patching, or have a new feature to contribute. ...
+You can get your local working copy of the [latest and greatest code](download.html "Download") by following the directions in [Download](download.html "Download") page. Review the To Do list in the [issue tracker](https://issues.apache.org/Jira/browse/SHIRO) and then choose a task that interests you.
```
```suggestion
You can get your local working copy of the [latest and greatest code](download.html "Download") by following the directions in [Download](download.html "Download") page. Review the To Do list in the [issue tracker](https://issues.apache.org/jira/browse/SHIRO) and then choose a task that interests you.
```

bdemers commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
+## Procedure for Reporting Bugs and Issues and Enhancement Suggestions
 
-If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/jira/browse/SHIRO) first as it may be already reported.
+If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/Jira/browse/SHIRO) first as it may be already reported.
```
```suggestion
If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/jira/browse/SHIRO) first as it may be already reported.
```

bmarwell replied:
```suggestion
If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/jira/browse/SHIRO) first as it may be already reported.
```

no capital 'j', or dead link.

bdemers commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
+If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/Jira/browse/SHIRO) first as it may be already reported.
 
-The [Apache Shiro Issue Tracker](https://issues.apache.org/jira/browse/SHIRO) collates our known issues. Obviously not every issue is listed there. Some issues have been discussed on the mailing list but do not yet have an issue recorded.
+The [Apache Shiro Issue Tracker](https://issues.apache.org/Jira/browse/SHIRO) collates our known issues. Obviously not every issue is listed there. Some issues have been discussed on the mailing list but do not yet have an issue recorded.
```
```suggestion
The [Apache Shiro Issue Tracker](https://issues.apache.org/jira/browse/SHIRO) collates our known issues. Obviously, not every issue is listed there. Some issues have been discussed on the mailing list but do not yet have an issue recorded.
```

bmarwell replied:
```suggestion
The [Apache Shiro Issue Tracker](https://issues.apache.org/jira/browse/SHIRO) collates our known issues. Obviously not every issue is listed there. Some issues have been discussed on the mailing list but do not yet have an issue recorded.
```

```suggestion
If you think that you have found a bug or you have a suggestion for improvement, then please discuss it on one of the [mailing lists](mailing-lists.html "Mailing Lists"). However, please check our [issue tracker](https://issues.apache.org/jira/browse/SHIRO) first as it may be already reported.
```

no capital 'j', or dead link.

bdemers commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
+### Issue Tracker
 
-As a rule-of-thumb, discuss an issue on the developers mailing list first to work out any details. After it is confirmed to be worthwhile, and you are clear about it, then submit the bug description or patch via Bug Tracking.
+Experienced developers use the [issue tracker](https://issues.apache.org/Jira/browse/SHIRO) directly, as they are very sure when they have found a bug and when not. Less experienced users should first discuss it on the user or developer mailing list (as appropriate). Impatient people always enter everything into the issue tracker without caring if it is a bug of Shiro, or their own installation/c ...
```
```suggestion
Experienced developers use the [issue tracker](https://issues.apache.org/jira/browse/SHIRO) directly, as they are very sure when they have found a bug and when not. Less experienced users should first discuss it on the user or developer mailing list (as appropriate). Impatient people always enter everything into the issue tracker without caring if it is a bug of Shiro, or their own installation/configuration mistake - please do not do this.
```

bmarwell replied:
```suggestion
Experienced developers use the [issue tracker](https://issues.apache.org/jira/browse/SHIRO) directly, as they are very sure when they have found a bug and when not. Less experienced users should first discuss it on the user or developer mailing list (as appropriate). Impatient people always enter everything into the issue tracker without caring if it is a bug of Shiro, or their own installation/configuration mistake - please do not do this.
```

bdemers commented on how-to-contribute.md:
````diff
@@ -1,177 +1,198 @@
+Grab the Shiro source:
 
 ``` bash
 $ git clone https://gitbox.apache.org/repos/asf/shiro.git
````
```suggestion
$ git clone https://gitbox.apache.org/repos/asf/shiro.git
$ cd shiro
```

bdemers commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
-Once you've submitted your pull request, you should receive a response within a few days. If you receive no response within a week, please ping the shiro-dev mailing list (dev@shiro.apache.org).
+Non-committers have to submit patches for review. Apache Shiro accepts GitHub pull requests. 
+
+Apache Shiro has a read-only mirror on GitHub that is kept in sync with the canonical Git repo maintained by the Apache Software Foundation. For detailed instructions see the [GitHub Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md).
```
```suggestion
Apache Shiro has a mirror on GitHub that is kept in sync with the canonical Git repo maintained by the Apache Software Foundation. For detailed instructions see the [GitHub Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md).
```
(It's no longer read-only)

bdemers reviewed (approved):
Thanks @pitjazz!
I pushed a commit to fix some of the "jira" find/replace URLs that were introduced, you may want to take a quick look at them.

bdemers commented:
@pitjazz you may need to sign the CLA for his one (if you haven't already) https://apache.org/licenses/contributor-agreements.html

pitjazz commented:
> @pitjazz you may need to sign the CLA for his one (if you haven't already) https://apache.org/licenses/contributor-agreements.html

Ok, I'll take a look at it

pitjazz commented:
> @pitjazz you may need to sign the CLA for his one (if you haven't already) https://apache.org/licenses/contributor-agreements.html

@bdemers, signed CLA contract is now sent to Apache's secretary.

fpapon commented:
@pitjazz can you squash the commits please?

pitjazz commented:
> @pitjazz can you squash the commits please?

Rebase done: commits squashed as requested. Ping @fpapon

fpapon commented:
@bmarwell is it ok for you?

bmarwell commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
+You can get your local working copy of the [latest and greatest code](download.html "Download") by following the directions in [Download](download.html "Download") page. Review the To Do list in the [issue tracker](https://issues.apache.org/Jira/browse/SHIRO) and then choose a task that interests you.
 
-Document writers are usually the most wanted people so if you like to help but you're not familiar with the innermost technical details, don't worry: you can still be tremendously helpful!
+Perhaps you have noticed something that needs patching or have a new feature to contribute. Make the changes, do the testing, generate a patch, and discuss on the [dev mailing list](mailing-lists.html "Mailing Lists"). The process is easy and explained below. Subscribe to the mailing list!
```
Add a real link here (https://shiro.apache.org/mailing-lists.html)

pitjazz replied:
Fixed this.

But... The real link is missing approx from twenty *.md files from shiro-site project. Only mailing.lists.html is used on those files instead of the real link. FYI @bmarwell .

bmarwell commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
 <a name="HowtoContribute-help"></a>
 <a name="HowtoContribute-HelpWantedHere"></a>
-##Help Wanted Here
+## Help Wanted Here
```
```suggestion
## Help Wanted
```

bmarwell commented on how-to-contribute.md:
```diff
@@ -1,177 +1,198 @@
+* Providing new use-cases and requirements.
+** If you think that Shiro does not quite meet your needs then tell us about it on the mailing list.
+* Specifying, analyzing or designing new features - and beyond.
+**	If you wish to get further involved with this, please join the [`shiro-dev` mailing list](mailing-lists.html "Mailing Lists"), install and try out Shiro and read some of the [mail archives](mailing-lists.html "Mailing Lists").
```
https://shiro.apache.org/mailing-lists.html

bmarwell commented on how-to-contribute.md:
````diff
@@ -1,177 +1,198 @@
 ```
 
-You'll grab the Shiro source with git:
+Grab the Shiro source:
````
```suggestion
Obtain a copy of the Shiro source code:
```

pitjazz commented:
Committed change suggestions. Fixed merge conflicts. Squashed commits. Left a comment about mailing-lists URL to this pull request.

@bmarwell, @fpapon , ready for merge?

fpapon commented:
@pitjazz sorry for the delay
I can see a  .DS_Store file in the changes, can you remove it?

bmarwell commented:
It has been a while, and I started working on a `jbake` version: https://github.com/bmarwell/shiro-site/tree/jbake/jbake

How do we proceed?

bdemers commented:
@bmarwell I think these changes look good, the only issue is the `.DS_Store`

Looks like this needs to be merged manually (to delete the `.DS_Store` file),  I'll volunteer to do it, i just don't want to get in your way for your `jbake` work

fpapon commented:
> @bmarwell I think these changes look good, the only issue is the `.DS_Store`
> 
> Looks like this needs to be merged manually (to delete the `.DS_Store` file), I'll volunteer to do it, i just don't want to get in your way for your `jbake` work

I already asked to the user to remove the `.DS_Store` file but without answer so +1 to move forward manually.

bdemers commented:
I was able to delete the file and push the change to this PR.  I just wasn't able to do it through the GitHub UI 🤷

bdemers commented:
Thanks @pitjazz !!

bmarwell commented:
> I was able to delete the file and push the change to this PR.  I just wasn't able to do it through the GitHub UI 🤷 


Hi Brian,

can you create a PR for my jbake branch in bmarwell/shiro-site from this? Most files will resolve/merge just fine. I just ported over a few news entries and the 10 minute tutorial.

Thanks in advance! 😊
